src/codegen/unwinding.cpp

```diff
@@ -572,7 +577,83 @@ BoxedDict* getLocals(bool only_user_visible, bool includeClosure) {
                         const auto& locs = e.locations;
 
                         assert(locs.size() == 1);
-                        uint64_t v = frame_info.readLocation(locs[0]);
+                        uint64_t v = frame_iter.readLocation(locs[0]);
+                        if ((v & 1) == 0)
+                            is_undefined.insert(p.first.substr(12));
+
+                        break;
+                    }
+                }
+            }
+
+            for (const auto& p : cf->location_map->names) {
+                if (p.first[0] == '!'
```

Comment:
Hmm the way I was thinking about it is that it's just an optimization to store them in the ASTInterpreter instead of in the symbol table -- not sure how much it gains us but it seems reasonable?  And I think we do ok about making sure to manually include those any time we want to iterate over the symbol table.